feat: Rework Configurtion documentation for new Common Config (#978)
edgexfoundry/edgex-docs: docs_src/microservices/configuration/ConfigurationAndRegistry.md

## Configuration

Please refer to the following EdgeX Foundry ADRs for details (and design decisions) behind the configuration in EdgeX

> - [Sevice Self Config Seeding](../../../design/adr/0005-Service-Self-Config)

Reviewer: ```suggestion
- [Service Self Config Seeding](../../../design/adr/0005-Service-Self-Config)
```

- [Common Configuration](../../../design/adr/0026-Common%20Configuration/)

### Common Configuration

!!! edgey - "EdgeX 3.0"
    Common configuration in single location is new in Edgex 3.0

> Many of each EdgeX service's configuration settings are the same as all other services. 

Reviewer 2: I think the word each could be removed 

```suggestion
Many of EdgeX service's configuration settings are the same as all other services. 
```

Author: fixed

> These common configuration settings have been consolidating into a single common configuration location which is seeded by the **core-common-config-bootstrapper** service.

Reviewer 2: This reads better this way, in my opinion

```suggestion
These common configuration settings have been consolidated into a single common configuration location which is seeded by the **core-common-config-bootstrapper** service.
```

Author: fixed

This service seeds the configuration provider with the common configuration from its local file located in the `cmd/res/configuration.yaml`.
See the [Common Configuration](../CommonConfiguration/) for list of all the common configuration settings.

### Local Configuration

Because EdgeX Foundry may be deployed and run in several different ways, 
it is important to understand how configuration is loaded and from where it is sourced. 

> Referring to the cmd directory within the [edgex-go repository](https://github.com/edgexfoundry/edgex-go) , each service has its own folder. 

Reviewer: ```suggestion
Referring to the cmd directory within the [edgex-go repository](https://github.com/edgexfoundry/edgex-go), each service has its own folder. 
```

> Inside each service folder there is a `res` directory (short for "resource").

Reviewer: ```suggestion
Inside each service's folder there is a `res` directory (short for "resource").
```

> There you will find the configuration files in [TOML format](https://github.com/toml-lang/toml) that defines each service's configuration. 

Reviewer: Are we writing in third person for the docs?
```suggestion
There the configuration files in [TOML format](https://github.com/toml-lang/toml) define each service's configuration. 
```
